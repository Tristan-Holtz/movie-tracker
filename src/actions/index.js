export const addMovies = movies => ({
  type: 'ADD_MOVIES',
  movies
});

export const pageUp = page => ({
  type: 'PAGE_UP',
  page
});

export const pageDown = page => ({
  type: 'PAGE_DOWN',
  page
});

export const addToWatchList = toWatch => ({
  type: 'ADD_FAVORITE',
  toWatch
})
