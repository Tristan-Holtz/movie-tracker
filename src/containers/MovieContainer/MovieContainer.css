.movie-container {
  /* width: 100%; */
  display: flex;
  flex-wrap: wrap;
  padding-top: 30px;
}