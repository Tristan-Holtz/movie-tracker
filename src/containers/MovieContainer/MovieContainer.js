import React from 'react';
import { connect } from 'react-redux'
import { fetchMovies } from '../../apiCalls/apiCalls'
import MovieCard from '../../components/MovieCard/MovieCard';
import { addMovies, pageUp, pageDown } from '../../actions/index';
import { Link } from 'react-router-dom';

export const MovieContainer = ({ movies, page, pageUp, pageDown, addMovies }) => {
  const movieCards = movies.map(movie => {
    if(movie.genre_ids.includes(27||53)) {
      return <MovieCard {...movie} key={movie.id} />
    }
  })

  const forward = page => {
    pageUp(page)
    getMovies(page + 1)
  }
  
  const backward = page => {
    pageDown(page)
    getMovies(page - 1)
  }

  const getMovies = async page => {
    const movies = await fetchMovies(page)
    addMovies(movies)
  }

  return ( 
    <div>
      {movieCards}
      {page > 1 && 
      <Link to={`movies/page/${page - 1}`} onClick={() => backward(page)}>{page - 1}</Link>}
      <span>{page}</span>
      <Link to={`movies/page/${page + 1}`} onClick={() => forward(page) }>{page + 1}</Link>
    </div>
  )
}

export const mapStateToProps = state => ({
  movies: state.movies,
  page: state.page
})

export const mapDispatchToProps = dispatch => ({
  addMovies: movies => dispatch( addMovies(movies) ),
  pageUp: page => dispatch( pageUp(page) ),
  pageDown: page => dispatch( pageDown(page) )
})

export default connect(mapStateToProps, mapDispatchToProps)(MovieContainer)