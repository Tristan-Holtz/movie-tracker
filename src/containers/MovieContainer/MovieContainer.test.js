import React from 'react';
import { shallow } from 'enzyme';
import { addMovies, pageUp, pageDown } from '../../actions/index'
import { MovieContainer, mapStateToProps, mapDispatchToProps } from './MovieContainer'

describe('MovieContainer', () => {
  let wrapper, mockProps
  beforeEach(() => {
    mockProps = {
      page: 1,
      movies: [
        {
            "genre_ids": [
                27,
                9648,
                878,
                53
            ],
            "id": 555974,
            "title": "The Invisible Man",
            "overview": "When Cecilia's abusive ex takes his own life and leaves her his fortune, she suspects his death was a hoax. As a series of coincidences turn lethal, Cecilia works to prove that she is being hunted by someone nobody can see.",
            "release_date": "2020-02-26"
        },
        {
            "genre_ids": [
                27,
                9648,
                53
            ],
            "id": 570670,
            "title": "Brahms: The Boy II",
            "overview": "After a family moves into the Heelshire Mansion, their young son soon makes friends with a life-like doll called Brahms.",
            "release_date": "2020-02-20"
        }
      ]
    }

    wrapper = shallow(<MovieContainer {...mockProps} />)
  })

  it('should match a snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })
  describe('mapStateToProps', () => {
    it('should return an object with a page number and an array of movies', () => {
      const mockState = {
        page: 1,
        movies: [
          {
              "genre_ids": [
                  27,
                  9648,
                  878,
                  53
              ],
              "title": "The Invisible Man",
              "overview": "When Cecilia's abusive ex takes his own life and leaves her his fortune, she suspects his death was a hoax. As a series of coincidences turn lethal, Cecilia works to prove that she is being hunted by someone nobody can see.",
              "release_date": "2020-02-26"
          },
          {
              "genre_ids": [
                  27,
                  9648,
                  53
              ],
              "title": "Brahms: The Boy II",
              "overview": "After a family moves into the Heelshire Mansion, their young son soon makes friends with a life-like doll called Brahms.",
              "release_date": "2020-02-20"
          }
        ]
      }
  
      const expected = {
        page: 1,
        movies: [
          {
              "genre_ids": [
                  27,
                  9648,
                  878,
                  53
              ],
              "title": "The Invisible Man",
              "overview": "When Cecilia's abusive ex takes his own life and leaves her his fortune, she suspects his death was a hoax. As a series of coincidences turn lethal, Cecilia works to prove that she is being hunted by someone nobody can see.",
              "release_date": "2020-02-26"
          },
          {
              "genre_ids": [
                  27,
                  9648,
                  53
              ],
              "title": "Brahms: The Boy II",
              "overview": "After a family moves into the Heelshire Mansion, their young son soon makes friends with a life-like doll called Brahms.",
              "release_date": "2020-02-20"
          }
        ]
      }
  
      const mappedProps = mapStateToProps(mockState)
      expect(mappedProps).toEqual(expected)
    })
  })

  describe('mapDispatchToProps', () => {
    it('should call dispatch with an addMovies action when addMovies is called', () => {
      const mockDispatch = jest.fn();
      const movies = [
        {
            "genre_ids": [
                27,
                9648,
                878,
                53
            ],
            "title": "The Invisible Man",
            "overview": "When Cecilia's abusive ex takes his own life and leaves her his fortune, she suspects his death was a hoax. As a series of coincidences turn lethal, Cecilia works to prove that she is being hunted by someone nobody can see.",
            "release_date": "2020-02-26"
        },
        {
            "genre_ids": [
                27,
                9648,
                53
            ],
            "title": "Brahms: The Boy II",
            "overview": "After a family moves into the Heelshire Mansion, their young son soon makes friends with a life-like doll called Brahms.",
            "release_date": "2020-02-20"
        }
      ]
      const action = addMovies(movies)
      const mappedProps = mapDispatchToProps(mockDispatch)
  
      mappedProps.addMovies(movies);
      expect(mockDispatch).toHaveBeenCalledWith(action);
    })

    it('should call dispatch when pageDown is called', () => {
      const mockDispatch = jest.fn();
      const page = 1
      const action = pageDown(page)
      const mappedProps = mapDispatchToProps(mockDispatch)
  
      mappedProps.pageDown(page);
      expect(mockDispatch).toHaveBeenCalledWith(action);
    })

    it('should call dispatch when pageUp is called', () => {
      const mockDispatch = jest.fn();
      const page = 1
      const action = pageUp(page)
      const mappedProps = mapDispatchToProps(mockDispatch)
  
      mappedProps.pageUp(page);
      expect(mockDispatch).toHaveBeenCalledWith(action);
    })
  })
})