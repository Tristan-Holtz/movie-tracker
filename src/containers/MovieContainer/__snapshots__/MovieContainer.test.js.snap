// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieContainer should match a snapshot 1`] = `
<main>
  <div
    className="movie-container"
  >
    <Connect(MovieCard)
      genre_ids={
        Array [
          27,
          9648,
          878,
          53,
        ]
      }
      id={555974}
      key="555974"
      overview="When Cecilia's abusive ex takes his own life and leaves her his fortune, she suspects his death was a hoax. As a series of coincidences turn lethal, Cecilia works to prove that she is being hunted by someone nobody can see."
      release_date="2020-02-26"
      title="The Invisible Man"
    />
    <Connect(MovieCard)
      genre_ids={
        Array [
          27,
          9648,
          53,
        ]
      }
      id={570670}
      key="570670"
      overview="After a family moves into the Heelshire Mansion, their young son soon makes friends with a life-like doll called Brahms."
      release_date="2020-02-20"
      title="Brahms: The Boy II"
    />
  </div>
  <div
    className="button-container"
  >
    <span>
      1
    </span>
    <Link
      className="page-buttons"
      onClick={[Function]}
      to="movies/page/2"
    >
      2
    </Link>
  </div>
</main>
`;
